app/src/test/java/org/mozilla/fenix/components/metrics/AppStartupTelemetryTest.kt

```diff
@@ -0,0 +1,241 @@
+        setupIntentMock(APP_ICON)
+        launchApplicationAndPutApplicationInBackground()
+
+        appStartupTelemetry.onHomeActivityOnRestart()
+        appStartupTelemetry.onHomeActivityOnNewIntent(intent.toSafeIntent())
+        appStartupTelemetry.onHomeActivityOnResume()
+
+        val validMetric = AppAllStartup(APP_ICON, HOT, false)
+        verify(exactly = 1) { metricController.track(validMetric) }
+    }
+
+    @Test
+    fun `GIVEN, application  exists and is backgrounded, WHEN application is launched again through url link and HomeActivity is restarted THEN records the correct values`() {
+        setupIntentMock(LINK)
+        launchApplicationAndPutApplicationInBackground()
+
+        appStartupTelemetry.onHomeActivityOnRestart()
+        appStartupTelemetry.onHomeActivityOnNewIntent(intent.toSafeIntent())
+        appStartupTelemetry.onHomeActivityOnResume()
+
+        val validMetric = AppAllStartup(LINK, HOT, false)
+        verify(exactly = 1) { metricController.track(validMetric) }
+    }
+
+    @Test
+    fun `WHEN application is launched and onResume() is called twice THEN metric is reported only once`() {
+        setupIntentMock(LINK)
+        appStartupTelemetry.onExternalAppBrowserOnCreate(intent.toSafeIntent(), false)
+        appStartupTelemetry.onHomeActivityOnResume()
+
+        appStartupTelemetry.onHomeActivityOnResume()
+
+        verify(exactly = 1) { metricController.track(any()) }
+    }
+
+    @Test
+    fun `GIVEN application is in background WHEN application is launched again through unknown source and HomeActivity exists THEN records the correct values`() {
+        setupIntentMock(UNKNOWN)
+        launchApplicationAndPutApplicationInBackground()
+
+        appStartupTelemetry.onHomeActivityOnCreate(intent.toSafeIntent(), false)
+        appStartupTelemetry.onHomeActivityOnResume()
+
+        val validMetric = AppAllStartup(UNKNOWN, WARM, false)
+        verify(exactly = 1) { metricController.track(validMetric) }
+    }
+
+    @Test
+    fun `GIVEN application exists and is backgrounded WHEN application started again through app icon but HomeActivity is recreated from savedInstanceState THEN records the correct values`() {
+        setupIntentMock(APP_ICON)
+        launchApplicationAndPutApplicationInBackground()
+
+        appStartupTelemetry.onHomeActivityOnCreate(intent.toSafeIntent(), true)
+        appStartupTelemetry.onHomeActivityOnResume()
+
+        val validMetric = AppAllStartup(APP_ICON, WARM, true)
+        verify(exactly = 1) { metricController.track(validMetric) }
+    }
+
+    private fun launchApplicationAndPutApplicationInBackground() {
+        appStartupTelemetry.onFenixApplicationOnCreate()
+        appStartupTelemetry.onHomeActivityOnCreate(intent.toSafeIntent(), false)
+        appStartupTelemetry.onHomeActivityOnResume()
+
+        // have to clear the mock function calls so it doesnt interfere with tests
+        clearMocks(metricController, answers = false)
```

Comment:
What does answers = false do?

Reply:
I couldn't really find something official and clear but it's consistent so I went on to use it. basically, it is clearing up calls to metricController.
For example, 

```
metricController.track(something)
clearMocks(metricController, answers = false) 
metricController.track(anotherThing)

verify(exactly = 1) { metricController.track(any()) } // this will be true
```

Reply:
Do you need `answers = false` or does `clearMocks(metricController)` work? I would prefer we do the latter if it does and are not sure what the arguments do.

Reply:
we need the `answers` part. otherwise, we need to re-init the mock every time.